### DISCUSSION TITLE
Combination of options aspratio and morpho

### DISCUSSION THREAD

**Discussion by MarcJos** [neperfepx/neper#196] [Q&A] [answered]
Dear Romain Quey,
I am very puzzled for I do not understand the interaction of 2 options : aspratio and morpho "diameq".
Based on the manual complemented with [Quey, Renversade, 2018], I expect the following:
- `-morpho aspratio` is to compute the tessellation in a dilatated cube,
- `-morpho diameq` triggers the use of the Laguerre tessellation instead of the Voronoi one.
In other words, I expect x to be in cell of center c (which I think of beeing of coordinates described just after *seed in the .tess) it is minimizes
\sum_{j=1}^d \frac{|x_i - c_i|^2}{a_i^2} - w^2,
where a is the renormalized aspect ratio, and w stands for the weight* written in files .tess. (I noticed that aspratio is renormalized so that <aspratio> = 1.)
However, if I could check separately these two interpretations by independently building the tessellation and comparing with neper, these 2 together seem to interact in a way I cannot figure out, and the formula above seems not to be true.
Could you please describe it, by a formula as in [Quey, Renversade, 2018](1)?
With kind regards,

Marc Josien

* by reverse engineering, I found out that the fourth column of the *seed of .tess actually corresponds to sqrt{weigths} in [Quey, Renversade, 2018](1).

**Comment by rquey** [comment, marked as answer]
`-morpho aspratio` defines the aspect ratio of the cells. It is used independently of `-morpho diameq`. The aspect ratios are indeed normalized so that the 3 values multiply to 1 (they are ratios...). The method to get the expected aspect ratios for the cells is to 1/ apply the inverse transformation to the domain, 2/ generate the tessellation into equiaxed cells (as usual) and 3/ apply the aspect-ratio transformation to the whole tessellation. In this way, the domain is as expected and the cells have the expected aspect ratios.

For example, if the domain is the usual unit cube and the aspect ratios are 2:1:0.5, the domain is first scaled to a cube of dimensions 0.5 x 1 x 2, the tessellation is generated, and the tessellation is scaled by factors of 2:1:0.5. Therefore, the grains have the expected aspect ratios (2:1:0.5) and the domain is the unit cube.

`-morpho diameq` applies to the tessellation generated at step 2. It therefore does not interact with `aspratio`.

And yes, there are some inconsistencies with the seed weights. I've been reluctant to change this for backward compatibility reasons, but it may happen in a future major release.

**Comment by MarcJos** [comment]
Dear Romain Quey,
Thanks a lot for your answer which was very helpful! Indeed, I misunderstood the way you renormalized the aspratio, I took the arithmetic mean but not the geometric one.
With kind regards,


Marc

(Sorry for my previous message, which was not well-displayed, for I thought it would be displayed in plain text.)
